[GDAL] find package error

@dg0yt this might be more related to you. 


```
CMake Error at C:/Program Files/CMake/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find GDAL (missing: GDAL_LIBRARY) (found version "3.2.2")
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/CMake/share/cmake-3.21/Modules/FindGDAL.cmake:185 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  D:/dev/node-mapnik/build/vcpkg_installed/x64-windows/share/gdal/vcpkg-cmake-wrapper.cmake:28 (_find_package)
  D:/dev/node-mapnik/vcpkg/scripts/buildsystems/vcpkg.cmake:751 (include)
  plugins/input/ogr/CMakeLists.txt:3 (find_package)
```

In the end, it is this line `_find_package(${ARGS})`  in vcpkg-cmake-wrapper.cmake. 

If I comment it out, it all works fine and all libraries are discovered. 

PS: I couldn't figured it out why it complainted. If I add a `message(FATAL_ERROR ${GDAL_LIBRARY})` before `_find_package` I'm getting the path of the libraries. 
contents of `GDAL_LIBRARY`
```
optimized;D:/dev/node-mapnik/build/vcpkg_installed/x64-windows/lib/gdal_i.lib;debug;D:/dev/node-mapnik/build/vcpkg_installed/x64-windows/debug/lib/gdal_i_d.lib
```

--- a/ports/gdal/vcpkg-cmake-wrapper.cmake
+++ b/ports/gdal/vcpkg-cmake-wrapper.cmake
@@ -23,6 +23,8 @@ if(NOT GDAL_INCLUDE_DIR OR NOT GDAL_LIBRARY)
     message(FATAL_ERROR "Installation of vcpkg port gdal is broken.")
 endif()
 
+set(GDAL_LIBRARY "${GDAL_LIBRARY}" CACHE STRING "")
+
 set(FindGDAL_SKIP_GDAL_CONFIG TRUE)
 
 _find_package(${ARGS})

--- a/ports/gdal/vcpkg.json
+++ b/ports/gdal/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "gdal",
   "version-semver": "3.2.2",
-  "port-version": 2,
+  "port-version": 3,
   "description": "The Geographic Data Abstraction Library for reading and writing geospatial raster and vector data",
   "homepage": "https://gdal.org",
   "supports": "!arm",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2218,7 +2218,7 @@
     },
     "gdal": {
       "baseline": "3.2.2",
-      "port-version": 2
+      "port-version": 3
     },
     "gdcm": {
       "baseline": "3.0.7",

--- a/versions/g-/gdal.json
+++ b/versions/g-/gdal.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "5bb72450f69322e39e09bea191e5c947833e698d",
+      "version-semver": "3.2.2",
+      "port-version": 3
+    },
+    {
       "git-tree": "cee8a26ec45230cb4b11e18582b41291b264d82e",
       "version-semver": "3.2.2",
       "port-version": 2

